external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -100,9 +100,14 @@ case class AvroDataToCatalyst(
       case NonFatal(e) => parseMode match {
         case PermissiveMode => nullResultRow
         case FailFastMode =>
-          throw new SparkException("Malformed records are detected in record parsing. " +
+          val msg = if (e.getMessage != null) {
+            e.getMessage + "\n"
+          } else {
+            ""
+          }
+          throw new SparkException(msg + "Malformed records are detected in record parsing. " +
             s"Current parse Mode: ${FailFastMode.name}. To process malformed records as null " +
-            "result, try setting the option 'mode' as 'PERMISSIVE'.", e.getCause)
+            "result, try setting the option 'mode' as 'PERMISSIVE'.", e)
```

Comment:
Adding causes looks fine. I was actually wondering adding `msg` is necessary.

Reply:
I see. It would be like
```
org.apache.spark.SparkException: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('1' (code 49)): was expecting a colon to separate field name and value
 at [Source: (InputStreamReader); line: 1, column: 7]
Malformed records are detected in record parsing. Parse Mode: FAILFAST. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'.
	at org.apache.spark.sql.catalyst.util.FailureSafeParser.parse(FailureSafeParser.scala:80)
	at org.apache.spark.sql.catalyst.expressions.JsonToStructs.nullSafeEval(jsonExpressions.scala:594)
...
Caused by: org.apache.spark.sql.catalyst.util.BadRecordException: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('1' (code 49)): was expecting a colon to separate field name and value
 at [Source: (InputStreamReader); line: 1, column: 7]
...
```

Reply:
How about we just remove `msg` change? If I am not mistaken, usually it's just added at the cause alone.

Reply:
Agree, When I post the output, I have the same thought..Haha.